src/Interactive/Features/Interactive/Core/InteractiveHost.LazyRemoteService.cs

```diff
@@ -62,20 +62,23 @@ public void Dispose(bool joinThreads)
                     }
 
                     bool initializing = true;
-                    cancellationToken.Register(() =>
-                    {
-                        if (initializing)
+                    RemoteExecutionResult initializationResult;
+                    using (cancellationToken.Register(
+                        () =>
                         {
-                            // kill the process without triggering auto-reset:
-                            remoteService.Dispose(joinThreads: false);
-                        }
-                    });
-
-                    // try to execute initialization script:
-                    var initializationResult = await Async<RemoteExecutionResult>(remoteService, (service, operation) =>
+                            if (initializing)
+                            {
+                                // kill the process without triggering auto-reset:
+                                remoteService.Dispose(joinThreads: false);
+                            }
+                        }))
                     {
-                        service.InitializeContextAsync(operation, Options.InitializationFile, isRestarting: InstanceId > 1);
-                    }).ConfigureAwait(false);
+                        // try to execute initialization script:
+                        initializationResult = await Async<RemoteExecutionResult>(remoteService, (service, operation) =>
+                        {
+                            service.InitializeContextAsync(operation, Options.InitializationFile, isRestarting: InstanceId > 1);
+                        }).ConfigureAwait(false);
+                    }
 
                     initializing = false;
```

Comment:
Ohhhhhhhhh.  I see now.  Man, this was hard to parse.  I thought the part of the code in the using was actually in teh lambda.  

Can we pull the 'using' argument into a local variable to make it clearer what's going on?  Having a complex lambda in the expression part of a using-block is super hard to read.